base/sort.jl

```diff
@@ -2270,7 +2270,8 @@ uint_map(x::Signed, ::ForwardOrdering) =
 uint_unmap(::Type{T}, u::Unsigned, ::ForwardOrdering) where T <: Signed =
     xor(signed(u), typemin(T))
 
-UIntMappable(T::BitIntegerType, ::ForwardOrdering) = unsigned(T)
+UIntMappable(T::Type{<:Base.BitInteger}, ::ForwardOrdering) =
+    isconcretetype(T) ? unsigned(T) : nothing
```

Comment:
FYI: sorting assumes that all `Base.BitInteger`s support `unsigned`, `typemin`, and `xor`. And that once converted to an unsigned integer with `unsigned`, have a bunch of convenient properties that connect bitshift operations, masking, and ordering, and that bit operations are very efficient.